src/components/ContactsPage.js

```diff
@@ -79,30 +91,88 @@ class ContactsPage extends React.Component {
     const updatedContacts = contacts.filter(x => !x.isChecked);
     this.setState({ contacts: updatedContacts });
   }
-  
-  numberOfSelectedContacts() {
+
+  numberOfSelectedContacts = () => {
     const { contacts } = this.state;
     return contacts.filter(x => x.isChecked).length || null;
   }
 
-  changeIsChecked(id) {
+  changeIsChecked = (id) => {
     const { contacts } = this.state;
     const contact = contacts.find(item => item.id === id);
     contact.isChecked = !contact.isChecked;
     this.setState({ contacts });
   }
 
-  updateLikes(id, step) {
+  updateLikes = (id, step) => {
     const { contacts } = this.state;
     contacts.find(item => item.id === id).likes += step;
     this.setState({ contacts });
   }
 
-  deleteContact(id) {
+  deleteContact = (id) => {
     const { contacts } = this.state;
     const updatedContacts = contacts.filter(c => c.id !== id);
     this.setState({ contacts: updatedContacts });
   }
+
+  deleteContactPopUp = (id) => {
+    this.setState({
+      popoup: {
+        showPopup: true,
+        message: WARNING_MESSAGES.DELETE_ONE,
+        approveFlag: {
+          flag: APPROVE_FLAGS.DELETE_ONE,
+          contactId: id
```

Comment:
Add this 'contactId' at one level above. It has nothing to do with the flag. An maybe name it 'contactIdToDelete'.